### PR TITLE
[Access] Add API that return node information such as version, commit, spork Id and protocol version.

### PR DESCRIPTION
**Pull request by Guitarheroua**
https://github.com/onflow/flow-go/issues/2134

### Context

This pull request adds a new API query `GetNodeVersionInfo` for RPC/REST that returns essential node information, including version number, commit hash, spork ID, and protocol version.

As part of this PR:

* The protobuf/openapi structures have been extended (see https://github.com/onflow/flow/pull/1332)
* Tests for the new API have been added for RPC/REST.

NOTE:
The Cadence version is still under discussion. See https://github.com/onflow/flow-go/issues/2134#issuecomment-1485446548

### CODE REVIEW

File: engine/access/rest/models/node_version_info.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+	"strconv"
+
+	"github.com/onflow/flow-go/access"
+)
+
+func (t *NodeVersionInfo) Build(params *access.NodeVersionInfo) {
+	t.Semver = params.Semver
+	t.Commit = params.Commit
+	t.SporkId = params.SporkId.String()
+	t.ProtocolVersion = strconv.FormatUint(uint64(params.ProtocolVersion), 10)
```
Comment (peterargue) [inline engine/access/rest/models/node_version_info.go:13]: nit
```suggestion
	t.ProtocolVersion = util.FromUint64(params.ProtocolVersion)
```

**Review by peterargue — COMMENTED** [review]
Added a few small comments, but otherwise looks good.

File: engine/access/rest/node_version_info_test.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	})
+}
+
+func nodeVersionInfoExpectedStr(nodeVersionInfo *access.NodeVersionInfo) string {
+	return fmt.Sprintf(`{
+			"semver": "%s",
+			"commit": "%s",
+			"spork_id": "%s",
+            "protocol_version": "%s"
+		}`, nodeVersionInfo.Semver, nodeVersionInfo.Commit, nodeVersionInfo.SporkId.String(), strconv.FormatUint(nodeVersionInfo.ProtocolVersion, 10))
```
Comment (peterargue) [inline engine/access/rest/node_version_info_test.go:56]: nit
```suggestion
            "protocol_version": "%d"
		}`, nodeVersionInfo.Semver, nodeVersionInfo.Commit, nodeVersionInfo.SporkId.String(), nodeVersionInfo.ProtocolVersion)
```

**Review by peterargue — APPROVED** [review]
Looks good. When you update the protobuf version, please run `make tidy` to fix the lint/test errors

**Comment by peterargue** [comment]
@Guitarheroua not sure if you can see the CI results, but those tests are erroring during the container build:
```
Error: #17 1.911 cmd/observer/node_builder/observer_builder.go:1060:5: cannot use backend.NewNetworkAPI(node.State, node.RootChainID, backend.DefaultSnapshotHistoryLimit) (value of type *backend.backendNetwork) as type "github.com/onflow/flow-go/engine/protocol".NetworkAPI in argument to protocol.New:
#17 1.911 	*backend.backendNetwork does not implement "github.com/onflow/flow-go/engine/protocol".NetworkAPI (missing GetNodeVersionInfo method)
```

**Comment by peterargue** [comment]
@Guitarheroua can you fix those conflicts. I think they were from your other PR

**Review by koko1123 — COMMENTED** [review]
Holding off on a approving till the conflicts are fixed, looks good otherwise

**Review by koko1123 — APPROVED** [review]

**Comment by peterargue** [comment]
bors merge

**Pull request merged**